### ISSUE TITLE
Binary doesn't run on Mac M1 (silicon)

### ISSUE THREAD

**Issue by ngocphamm** [mitmproxy/mitmproxy#6590]
#### Problem Description
I apologize if this is somehow an expected outcome as I can see that there's only `x86_64` dmg file provided for macOS. I'm using macOS 14.2.1 on a M1 Max machine, and I cannot get mitmproxy to work. It's complaining about the CPU type.

I do not have Rosetta installed on my system.

#### Steps to reproduce the behavior:
1. Install using Homebrew `brew install --cask mitmproxy`
2. Try `mitmproxy --version`
3. See the error

```
==> Installing Cask mitmproxy
==> Linking Binary 'mitmproxy' to '/opt/homebrew/bin/mitmproxy'
==> Linking Binary 'mitmdump' to '/opt/homebrew/bin/mitmdump'
==> Linking Binary 'mitmweb' to '/opt/homebrew/bin/mitmweb'
🍺  mitmproxy was successfully installed!
~ »  mitmproxy --version
zsh: bad CPU type in executable: mitmproxy
```

#### System Information
* macOS version: `14.2.1`, no Rosetta

**Comment by mhils** [comment]
TL;DR: We currently require Rosetta unfortunately. 

Building on arm64 does work, but GitHub does not have usable macOS arm64 CI runners yet (the available ones only work for pro teams, which makes no financial sense for a FOSS project). So it's a tradeoff between release complexity and requiring Rosetta. So far it seems that Rosetta is widespread enough to be the lesser evil. But we'll surely provide arm64 builds once that becomes feasible.

**Comment by ngocphamm** [comment]
Thanks for a very swift response. I understand completely. I made the MR above to add to the documentation the requirement of Rosetta on Apple Silicon. Please let me know if that looks good enough, or even needed.

**Comment by sky96111** [comment]
> TL;DR: We currently require Rosetta unfortunately.
> 
> Building on arm64 does work, but GitHub does not have usable macOS arm64 CI runners yet (the available ones only work for pro teams, which makes no financial sense for a FOSS project - we'd need to pay monthly for every contributor). So it's a tradeoff between release complexity and requiring Rosetta. So far it seems that Rosetta is widespread enough to be the lesser evil. But we'll surely provide arm64 builds once that becomes feasible.

https://github.com/actions/runner-images/issues/9254
Seems new macOS 14 Arm64 runner is available for open source plan. Could you please reconsider packaging for apple silicon  now?

**Comment by mhils** [comment]
The latest release already has a macOS arm64 variant. :)
